Add support for multiple custom scores to the lasagne.NeuralNet class

alito opened the pull request:
The custom_score parameter seemed very useful, but I wanted more. This patch lets a list of ('name', scorer) pairs be passed through the custom_scores parameter.

Passing a single custom_score still works but handlers that run on_epoch_finished that expect self.custom_score to be a pair will most likely break.  (Externals to the library I mean, I modified PrintLog of course).

alito commented:
Added custom_score as a property so that it should work with almost any external code that used to work with library before patch

dnouri commented on nolearn/lasagne/base.py:
```diff
@@ -200,6 +200,13 @@ def __init__(
                 "It's unnecessary.")  # BBB
 
+        if 'custom_score' in kwargs:
+            # add it to custom_scores
```
Can you add a `DeprecationWarning` here as per the `if 'eval_size'` above?

dnouri commented on nolearn/lasagne/base.py:
```diff
@@ -631,3 +643,29 @@ def _get_param_names(self):
+
+    @property
+    def custom_score(self):
+        if self.custom_scores and len(self.custom_scores) == 1:
```
Appreciate the thoroughness, but I don't think we need to support backward compatibility for setting attributes on the `NeuralNet` object directly.  As far as I'm concerned, setting `net.custom_score` was never part of the API.

alito commented:
Added warning, and removed setter and deleter of custom_score. Left getter since I was following the strict letter of your comment but I can delete that too.

Now, regarding the conflicts, how is this usually handled? git rebase?

dnouri commented:
Yes, I think you can remove the getter as well.

Also `git rebase` will work fine, yes.

alito commented:
Done. I hope I did the rebase right. Diff looks fine to me, not sure about the history.

dnouri commented:
Thanks.  The commit log still looks too busy.  Can you squash all these commits into one?
